app/models/spree/order.rb

```diff
@@ -733,6 +733,7 @@ def skip_payment_for_subscription?
     end
 
     def require_customer?
+      return false if customer_creation == false
       return false if new_record? || state == 'cart'
```

Comment:
Isn't the order in cart state when it's at the "customer creation" step? Or is it later? Address state?

Reply:
The one just after, address, exactly. All this machinery we have is to implement that "conditional"

Reply:
I'm just wondering if we should restrict it a bit, like we're now unconditionally setting `@order.customer_creation = false` in the edit action before advancing the order steps with `OrderWorkflow`, right? And that'll mean that `#require_customer?` will always return `false`, regardless of the order state? Or no?

Would this be preferable in the edit action? :point_down: 
```ruby
@order.customer_creation = false if @order.address?
```
Sorry for all the questions! I'm just trying to make sense of it in my head :thinking:

Reply:
This is not needed now with the change in approach, because nothing blows up if the customer fails to be created, but you do well to ask.

At the point where I just added an item in the cart :point_down: , a request is sent to save it. First, `#requier_customer?` is checked and it does return true. That's because `new_record? ==> false` and `state == "address"`. This added `customer_creation` flag avoided this.

Reply:
Wait, now we're not doing this at all? I'm confused :see_no_evil: I need to drink more coffee and re-read it :sweat_smile:

Reply:
dude! you made me change the approach and simply notify Bugsnag :sweat_smile: :joy:

Reply:
Ah, I assumed we would do the "soft" Bugnsag thing, but keep the `@order.customer_creation = false` idea, so the order can progress to the next step? :joy:

Reply:
Now I see. This is the misunderstanding. It does progress to the next step right now, but leaves the order without an associated customer up until the next page, where the user fills up the customer form.